app/controllers/application_controller.rb

```diff
@@ -285,7 +285,7 @@ def ensure_logged_in
     end
 
     def redirect_to_login_if_required
-      redirect_to :login if SiteSetting.login_required? && !current_user
+      redirect_to(:login, host: Discourse.current_hostname) if SiteSetting.login_required? && !current_user
```

Comment:
I don't know if this will actually fix the problem. What is `Discourse.current_hostname`? Is that a setting or is it derived from the request? In errbit the hostname is `http://robots.txt`, which makes me wonder if something is crawling the site by ip address /shrug